src/EscapeCodeGenerator.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "Defs.hpp"
+#include <string>
+
+struct EscapeCodeGenerator {
+
+    using EscapeCode = std::string;
+
+    virtual EscapeCode cursorForward(int) { return EscapeCode{}; }
+    virtual EscapeCode cursorBackward(int) { return EscapeCode{}; }
+    virtual EscapeCode clearTerminalLine() { return EscapeCode{}; }
+    virtual EscapeCode setColor(Color) { return EscapeCode{}; }
+
```

Comment:
inheriting from something and then `Derived * d = new Base; delete d;` is undefined behavior. Maybe, to prevent this, let's add virtual d-tor?
source: §5.3.5[expr.delete]/3